File: vom-server/src/main/java/vom/server/chaser/dto/SystemStats.java
package vom.server.chaser.dto;

import javax.persistence.Entity;
import javax.persistence.Id;

//@Entity
public class SystemStats {

  @Id
  private Long id;

  @Id
  private Long collected;

  private Double cpu;

  private Long diskFree;
  private Long diskTotal;

  private Long memoryFree;
  private Long memoryTotal;

  private Long networkTotal;
  private Long networkReceived;
  private Long networkSent;

//  private Double temperature;
//  private Double voltage;

}

File: vom-server/src/main/kotlin/vom/server/collector/stats/SystemStats.kt
package vom.server.collector.stats

import javax.persistence.*

@Entity
@Table(name = "SYSTEM_STATS")
@IdClass(SystemStatsId::class)
class SystemStats(
  @Id var id: String,
  @Id var collected: Long
) {

  var cpu: Double = -1.0

  @Column(name = "DISK_FREE")
  var diskFree: Long = -1

  @Column(name = "DISK_TOTAL")
  var diskTotal: Long = -1

  @Column(name = "MEM_FREE")
  var memoryFree: Long = -1

  @Column(name = "MEM_TOTAL")
  var memoryTotal: Long = -1

  @Column(name = "NET_RECV")
  var networkReceived: Long = -1

  @Column(name = "NET_SENT")
  var networkSent: Long = -1

  @Column(name = "NET_BW")
  var networkBandwidth: Long = -1

}
